workschedules

anisometropie opened the pull request:
- Display the workschedules in the stdcm debug interface.
- Add the new workschedule layer in the spacetime chart. 

### testing this branch
You need to create some workschedules manually.

For example here, taking these 2 track_sections in grenoble-lyon

- `62967baa-6667-11e3-81ff-01f464e0362d`
- `62961b24-6667-11e3-81ff-01f464e0362d`

`POST /work_schedules` :
```
{
    "work_schedule_group_name": "grenoble-lyon",
    "work_schedules": [
        {
            "start_date_time": "2024-10-10T19:00:00",
            "end_date_time": "2024-10-10T19:30:00",
            "track_ranges": [
                {
                    "track": "62967baa-6667-11e3-81ff-01f464e0362d",
                    "begin": 0,
                    "end": 5000.0
                }
            ],
            "obj_id": "work_schedule_obj_id",
            "work_schedule_type": "TRACK"
        },
                {
            "start_date_time": "2024-10-10T19:00:00",
            "end_date_time": "2024-10-10T19:30:00",
            "track_ranges": [
                {
                    "track": "",
                    "begin": 10000.0,
                    "end": 20000.0
                }
            ],
            "obj_id": "work_schedule_obj_id",
            "work_schedule_type": "TRACK"
        }
    ]
}
```
then update the stdcm search with `POST /stdcm/search_environment`

```
{
    "infra_id": 2,
    "electrical_profile_set_id": null,
    "work_schedule_group_id": ___, ====> the id of the group schedule you just created
    "search_window_begin": "2024-10-01T00:00:00",
    "search_window_end": "2024-10-31T00:00:00",
    "timetable_id": 3
}
```

![Screenshot_20241021_171934](https://github.com/user-attachments/assets/64438389-e4dd-4e5a-a7ed-f57dcf4e8779)


closes [#8643](https://github.com/OpenRailAssociation/osrd/issues/8643)

anisometropie committed:
front: display work schedules on space time chart

Signed-off-by: Valentin Chanas <anisometropie@gmail.com>

emersion commented:
This adds the work schedules to the STDCM debug mode only. Sounds like the mockup also displays these on the operational studies page. Also, there is a checkbox in the space time chart settings to toggle them.

Are these items out-of-scope, or not yet implemented, or something else?

anisometropie commented:
> This adds the work schedules to the STDCM debug mode only. Sounds like the mockup also displays these on the operational studies page. Also, there is a checkbox in the space time chart settings to toggle them.
> 
> Are these items out-of-scope, or not yet implemented, or something else?

@emersion Yes, this is only meant to exist on the STDCM side for now

Akctarus reviewed (approved):
LGTM

anisometropie committed:
front: fix scenario explorer update stdcm environment

Selecting a new scenario in debug interface no longer removes the work schedule group id.

Now also uses the electrical profile set from the scenario

Signed-off-by: Valentin Chanas <anisometropie@gmail.com>

SharglutDev reviewed (approved):
Lgtm and tested, good job !
